File: new-rss-reader/app/api/feeds/[id]/route.ts
import { type NextRequest, NextResponse } from "next/server"
import { prisma } from "@/lib/prisma"

export async function DELETE(request: NextRequest, { params }: { params: Promise<{ id: string }> }) {
  try {
    const { id } = await params

    await prisma.feed.delete({
      where: { id },
    })

    return NextResponse.json({
      success: true,
      message: "Feed deleted successfully",
    })
  } catch (error) {
    console.error("Error deleting feed:", error)
    return NextResponse.json({ success: false, error: "Failed to delete feed" }, { status: 500 })
  }
}

File: new-rss-reader/app/api/feeds/[id]/refresh/route.rs
import { type NextRequest, NextResponse } from "next/server"
import { RSSService } from "@/lib/rss-service"

const rssService = new RSSService()

export async function POST(request: NextRequest,{ params }: { params: Promise<{ symbol: string }> }) {
  try {
    const { id } = await params

    const result = await rssService.refreshFeed(id)

    return NextResponse.json({
      success: true,
      message: `Refreshed feed: ${result.feed.title}`,
      newArticles: result.newArticlesCount,
    })
  } catch (error) {
    console.error("Error refreshing feed:", error)
    return NextResponse.json(
      { success: false, error: error instanceof Error ? error.message : "Failed to refresh feed" },
      { status: 500 },
    )
  }
}
